src/shapes/ActiveSelection.spec.ts

```diff
@@ -4,9 +4,211 @@ import { ActiveSelection } from './ActiveSelection';
+      angle: 0,
+      flipX: false,
+      flipY: false,
+      opacity: 1,
+      fillRule: 'nonzero',
+      paintFirst: 'fill',
+      globalCompositeOperation: 'source-over',
+      skewX: 0,
+      skewY: 0,
+      strokeUniform: false,
+      objects: clone.objects,
+      layoutManager: {
+        type: 'layoutManager',
+        strategy: 'fit-content',
+      },
+    };
+
+    expect(clone).toEqual(expectedObject);
+
+    expect(group).not.toBe(clone);
+    expect(group.getObjects()).not.toBe(clone.objects);
+    expect(group.getObjects()[0]).not.toBe(clone.objects[0]);
+  });
+
+  it('toObject without default values', () => {
+    const group = makeAsWith2Objects();
+    group.includeDefaultValues = false;
+
+    const clone = group.toObject();
+    const objects = [
+      {
+        version: version,
+        type: 'Rect',
+        left: 10,
+        top: -30,
+        width: 30,
+        height: 10,
+        strokeWidth: 0,
+      },
+      {
+        version: version,
+        type: 'Rect',
+        left: -40,
+        top: -10,
+        width: 10,
+        height: 40,
+        strokeWidth: 0,
+      },
+    ];
+    const expectedObject = {
+      version: version,
+      type: 'ActiveSelection',
+      left: 50,
+      top: 100,
+      width: 80,
+      height: 60,
+      objects: objects,
+    };
+
+    expect(clone).toEqual(expectedObject);
+  });
+
+  it('_renderControls', () => {
+    expect(ActiveSelection.prototype._renderControls).toBeTypeOf('function');
+  });
+
+  test('fromObject', async () => {
+    const group = makeAsWith2ObjectsWithOpacity();
+
+    expect(ActiveSelection.fromObject).toBeTypeOf('function');
+    const groupObject = group.toObject();
+
+    const newGroupFromObject = await ActiveSelection.fromObject(groupObject);
+    const objectFromOldGroup = group.toObject();
+    const objectFromNewGroup = newGroupFromObject.toObject();
+
+    expect(newGroupFromObject).toBeInstanceOf(ActiveSelection);
+
+    expect(objectFromOldGroup.objects[0]).toEqual(
+      objectFromNewGroup.objects[0],
+    );
+    expect(objectFromOldGroup.objects[1]).toEqual(
+      objectFromNewGroup.objects[1],
+    );
+
+    expect(objectFromOldGroup).toEqual(objectFromNewGroup);
+  });
+
+  it('get with locked objects', () => {
+    const group = makeAsWith2Objects();
+
+    expect(group.get('lockMovementX')).toBe(false);
+
+    // TODO activeGroup
+    // group.getObjects()[0].lockMovementX = true;
+    // assert.equal(group.get('lockMovementX'), true);
+    //
+    // group.getObjects()[0].lockMovementX = false;
+    // assert.equal(group.get('lockMovementX'), false);
+
+    group.set('lockMovementX', true);
+    expect(group.get('lockMovementX')).toBe(true);
+
+    // group.set('lockMovementX', false);
+    // group.getObjects()[0].lockMovementY = true;
+    // group.getObjects()[1].lockRotation = true;
+    //
+    // assert.equal(group.get('lockMovementY'), true);
+    // assert.equal(group.get('lockRotation'), true);
+  });
```

Comment:
i think this test isn't necessary anymore.

Reply:
> i think this test isn't necessary anymore.

removed